src/hw/aarch64.c

```diff
@@ -0,0 +1,28 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+#include <stdint.h>
+#include <math.h>
+
+/*
+ * __aarch64_nsec_per_tick --
+ *     Return the nsec/tick calculated using the cntfrq_el0 register.
+ */
+static uint64_t
+__aarch64_proc_freq_hz(void)
+{
+    uint64_t freq;
+    /*
+     * The Armv8-A documentation warns that on a WARM reset the register is set to "an
+     * architecturally UNKNOWN value". We assume the OS will take care of that scenario, but protect
+     * against div 0 out of an abundance of caution.
+     */
+    __asm__ volatile("\tmrs\t%0, cntfrq_el0\n" : "=r"(freq));
```

Comment:
According to the documentation, only the bottom 32 bits are the frequency (the higher 32 bits are reserved), so you should mask the result accordingly.

https://developer.arm.com/documentation/ddi0601/2022-03/AArch64-Registers/CNTFRQ-EL0--Counter-timer-Frequency-register